src/base/ApplicationDynamoDBTable.ts

```diff
@@ -201,10 +201,10 @@ export class ApplicationDynamoDBTable extends Resource {
     }
 
     const constructPrefix = `${
-      indexName ? indexName : dynamoDB.name
+      indexName ? indexName : dynamoDB.friendlyUniqueId
```

Comment:
curious - what kind of value does `friendlyUniqueId` output? and are we sure it's valid as an index name value?

Reply:
also - do we know the impact of changing an index name? (also wonder if we have any implementations that don't specify an `indexName` - maybe this will have zero impact on existing services?)

Reply:
this is a great call out. I missed that this is an index name and might cause it to be recreated. ill do an example app on the old version of tf modules and then do an upgrade and see what happens.

Reply:
@jpetto So i created a dynamodb in the old version with indexes. Added data to it, and then ran apply using the new version. Turns out the data was still in tact and the index just recreated itself like a database index. So seems like this is all safe!